python/ray/remote_function.py

```diff
@@ -97,6 +99,13 @@ def _remote(self,
         """An experimental alternate way to submit remote functions."""
         worker = ray.worker.get_global_worker()
         worker.check_connected()
+
+        if self._last_export_time < worker._last_shutdown_time:
```

Comment:
Also, was this fix needed to get some test working? If so, which one?

Reply:
Yeah, lots of tests will fail without this fix. Because when we pass in the test directory, pytest will first import all files. Then all the remote functions will only be exported to the first session.